src/alert.cpp

```diff
@@ -2136,28 +2136,32 @@ namespace libtorrent
 		return buf;
 	}
 
-	dht_live_nodes_alert::dht_live_nodes_alert(aux::stack_allocator& alloc
-		, sha1_hash const& nid
+	namespace
+	{
+	using nodes_slot = std::tuple<int, aux::allocation_slot, int, aux::allocation_slot>;
+
+	nodes_slot write_nodes(aux::stack_allocator& alloc
 		, std::vector<std::pair<sha1_hash, udp::endpoint>> const& nodes)
-		: node_id(nid)
-		, m_alloc(alloc)
 	{
+		int v4_num_nodes = 0;
+		int v6_num_nodes = 0;
+
 		for (auto const& n : nodes)
 		{
 			if (n.second.protocol() == udp::v4())
-				m_v4_num_nodes++;
+				v4_num_nodes++;
 #if TORRENT_USE_IPV6
 			else
-				m_v6_num_nodes++;
+				v6_num_nodes++;
 #endif
 		}
 
-		m_v4_nodes_idx = alloc.allocate(m_v4_num_nodes * (20 + 6));
-		m_v6_nodes_idx = alloc.allocate(m_v6_num_nodes * (20 + 18));
+		aux::allocation_slot const v4_nodes_idx = alloc.allocate(v4_num_nodes * (20 + 6));
+		aux::allocation_slot const v6_nodes_idx = alloc.allocate(v6_num_nodes * (20 + 18));
```

Comment:
does it make sense to allocate space for IPv6 nodes if ``TORRENT_USE_IPV6`` is not set?
I wonder if this was a deliberate decision or an oversight